examples/gcoap/client.c

```diff
@@ -141,18 +141,48 @@ static void _resp_handler(const gcoap_request_memo_t *memo, coap_pkt_t* pdu,
     }
 }
 
-static bool _parse_endpoint(sock_udp_ep_t *remote,
-                            const char *addr_str, const char *port_str)
+static bool _parse_endpoint(sock_udp_ep_t *remote, const char *addr_str, const char *port_str)
 {
+    bool is_ipv4 = false;
+    bool is_ipv6 = false;
+
+#ifdef SOCK_HAS_IPV6
```

Comment:
I wonder if you could implement this using `if (IS_ACTIVE(SOCK_HAS_IPV6))` at the C language level rather than using the preprocessor. If that would be possible (without major pain) that would be preferred.

Reply:
`_parse_endpoint` will be removed by #17870 anyway, but I'll have a look where it makes sense to use the macro instead.